silq/tools/circuit_tools.py

```diff
@@ -0,0 +1,162 @@
+from typing import Type, Union
+from pathlib import Path
+import numpy as np
+import regex as re
+pi = np.pi
+
+from pygsti.objects.circuit import Circuit
+
+
+def convert_circuit(circuit, target_type: Type = str):
+    # First convert all types to string
+    if isinstance(circuit, Circuit):
+        expanded_circuit = []
+        for gate in circuit.get_labels():
+            if gate.issimple():
+                expanded_circuit.append(gate.name)
+            elif str(gate) == '[]':
+                gate.append('Gi')
+            else:
+                raise RuntimeError(f'Cannot understand gate {gate}')
+
+    elif isinstance(circuit, str):
+        # Replace [] by identity gate
+        expanded_circuit = circuit.replace('[]', 'Gi')
+
```

Comment:
would be nice to add here option for Randomized Benchmarking circuits, where we need to remove qubit name, i.e. something like: 
expanded_circuit = circuit.replace(f'{q_name}', '')

thoughts, opinions?

Reply:
Can you please give a concrete example? I don't recall anymore how the circuit refers to specific qubit names.
just on first thought I think as soon as there's more than 1 qubit it would be useful if the gates specify the relevant qubit

Reply:
Yes, so if you generate circuits in pygsti for Randomized Benchmarking, you need to define the name of the qubit and hence using 'Q' as an example will give you a circuit like this: "Gxmpi2:QGympi2:QGypi2:QGxpi2:Q@(Q)  0  0"
Obviously if we do for two or more qubits situation goes to Q1, Q2, etc.

Reply:
I think in the general case it is important to keep the qubit name, since you need a way to specify what qubit a pulse should be applied to when there is more than one qubit. 

When there is only one qubit, you could argue that the `":Q"` is unnecessary. We could add a renaming to remove this. But if we implement this, there are two complications:
1. It should also check if the pulse with `":Q"` exists, because the user might decide to add that at the end of the pulse name.
2. If we remove `":Q"`, we need to be careful that we only do this if no number follows the `Q`, because otherwise it ignores what qubit the pulse should be applied to.

One more thing to note is that if you define your pulse in `settings.pulses` as e.g. `"Gxmpi2:Q"` everything should be fine. So this implementing this is not a necessity, though it could be nice. At the moment I don't have time to add this. If anyone else is keen please feel free to do so. Otherwise I suggest we change this to an Issue, and add it in a later PR.

P.S. defining gates in a circuit with `:Q` is not just specific to RB, but also to the newer GST implementations

Reply:
I like the idea of defining gates/pulses as "Gxmpi2:Qn", where n can be anything. This looks quite general and there is no need to complicate things, just need to adhere everyone to one convention